e2-studio-star-rx72n-firmware/libs/rx_spi_link/src/rx_spi_link.c

```
@@ -0,0 +1,731 @@
+
+  /* CRITICAL: Get TX sequence BEFORE sending to avoid race.
+     * rx_session_get_tx() returns the NEXT sequence to be used.
+     * rx_spi_comm_send() will use this sequence and then increment.
+     * This approach is safe ONLY if rx_spi_link_send() is not called
+     * concurrently (enforced by caller via external mutex or task priority).
+     *
+     * Alternative solution: Modify rx_spi_comm_send() to return the
+     * sequence actually used, but that requires API change across codebase. */
+  uint16_t next_seq = 0;
+  (void)rx_session_get_tx(link->spi_handle->session, &next_seq);
+  const uint16_t expected_seq = next_seq; /* This will be used by send */
```

Clang (14.0.6)

[warning] variable 'expected_seq' is not initialized

(cppcoreguidelines-init-variables)